kusto/kcsb.go

```diff
@@ -0,0 +1,217 @@
+package kusto
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+type ConnectionStringBuilder struct {
+	clusterURI  string
+	authType    string
+	resourceURI string
+	authParams  map[string]interface{}
+	cloudInfo   CloudInfo
+}
+
+// environmental variables
+const (
+	tenantIdEnvVariable     = "AZURE_TENANT_ID"
+	clientIdEnvVariable     = "AZURE_CLIENT_ID"
+	clientSecretEnvVariable = "AZURE_CLIENT_SECRET"
+)
+
+// params mapping
+const (
+	tenantIDStr      string = "TenantID"
+	clientIDStr      string = "ClientID"
+	clientSecretStr  string = "ClientSecret"
+	appCertStr       string = "ApplicationCertificates"
+	appCertKeyStr    string = "ApplicationCertificateKey"
+	usernameStr      string = "Username"
+	userPasswordStr  string = "UserPassword"
+	appTokenStr      string = "ApplicationToken"
+	clientOptionsStr string = "ClientOptions"
+	sendCertChainStr string = "SendCertificateChain"
+	managedIDStr     string = "ManagedIdentityID"
```

Comment:
We support this capability (mi from connection string) due to security reasons.
Same for certificate subject name and certificate thumbprint